WYPopoverController/WYPopoverController.m

```diff
@@ -1768,23 +1749,21 @@ - (void)presentPopoverFromRect:(CGRect)aRect
 
     _backgroundView = [[WYPopoverBackgroundView alloc] initWithContentSize:contentViewSize];
     _backgroundView.appearing = YES;
-
-    _backgroundView.delegate = self;
     _backgroundView.hidden = YES;
 
-    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:_backgroundView action:@selector(tapOut)];
+    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleBackgroundTap:)];
     tap.cancelsTouchesInView = NO;
     [_overlayView addGestureRecognizer:tap];
 
     if (self.dismissOnTap) {
-      tap = [[UITapGestureRecognizer alloc] initWithTarget:_backgroundView action:@selector(tapOut)];
+      UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleBackgroundTap:)];
       tap.cancelsTouchesInView = NO;
       [_backgroundView addGestureRecognizer:tap];
     }
     
     if (self.dismissOnPassthroughViewTap) {
       for (UIView *view in self.passthroughViews) {
-        UITapGestureRecognizer *passthroughViewTap = [[UITapGestureRecognizer alloc] initWithTarget:_backgroundView action:@selector(tapOut)];
+        UITapGestureRecognizer *passthroughViewTap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleBackgroundTap:)];
```

Comment:
I think there still needs to be logic to remove these gesture recognizers from the passthroughViews on dealloc, since those views outlive the popoverController.